Classes/Service/QbankService.php

```diff
@@ -189,6 +219,235 @@ public function removeMediaUsageInFileReference(int $fileReferenceId): void
+                $file,
+                $targetProperty,
+                PropertyUtility::convertQbankToFileProperty(
+                    $qbankPropertyValue,
+                    $targetProperty
+                )
+            ));
+        }
+
+        $this->updateFileRecord($file->getUid(), false, true, $qbankId);
+
+        $this->eventDispatcher->dispatch(new AfterFilePropertyChangesEvent($file));
+    }
+
+    /**
+     * Synchronize file content for a particular file UID.
+     *
+     * @param int $fileId The FAL file UID
+     * @throws \TYPO3\CMS\Core\Resource\Exception\FileDoesNotExistException
+     */
+    public function replaceLocalMedia(int $fileId): void
+    {
+        // fetch current file
+        $file = $this->resourceFactory->getFileObject($fileId);
+        if ($file === null) {
+            throw new \TYPO3\CMS\Core\Resource\Exception\FileDoesNotExistException(
+                'No file found for given UID: ' . $fileId,
+                1623070299
+            );
+        }
+
+        // get qbank id
+        $mediaIdentifier = $this->getQbankMediaIdentifierForFile($fileId);
+
+        // get replacement qbank id
+        $replacedByMediaIdentifier = $this->getQbankReplacedByMediaIdentifierForFile($fileId);
+
+        // get replacement file
+        $replacedByFile = $this->createLocalMediaCopy($replacedByMediaIdentifier);
+        if ($replacedByFile === null) {
+            throw new \TYPO3\CMS\Core\Resource\Exception\FileDoesNotExistException(
+                'No file found for given QBank id: ' . $replacedByMediaIdentifier,
+                1623306399
+            );
+        }
+
+        // fetch all sys_file_references
+        $sysFileReferences = GeneralUtility::makeInstance(SysFileReferenceRepository::class)
+            ->fetchRawSysFileReferencesByFileId($fileId);
+
+        $data = [];
+        $cmd = [];
+        $pids = [];
+
+        // go through references to build data for sys_file_reference updates
+        if (is_array($sysFileReferences)) {
+            foreach ($sysFileReferences as $sysFileReference) {
+                // remove media usage for current file
+                $this->removeMediaUsageInFileReference($sysFileReference['uid']);
+
+                // create new unique id
+                $newId = StringUtility::getUniqueId('NEW');
+                // "copy" current reference to new
+                $data['sys_file_reference'][$newId] = $sysFileReference;
+                // unset uid on new
+                unset($data['sys_file_reference'][$newId]['uid']);
+                // change file id on new
+                $data['sys_file_reference'][$newId]['uid_local'] = $replacedByFile->getProperty('uid');
+
+                // remember pages
+                $pids[] = $sysFileReference['pid'];
+
+                // set current relation to deleted
+                $cmd['sys_file_reference'][$sysFileReference['uid']]['delete'] = 1;
```

Comment:
I think this is self-explanatory, so it shouldn't be necessary to comment each line.

```suggestion
                $this->removeMediaUsageInFileReference($sysFileReference['uid']);

                $newId = StringUtility::getUniqueId('NEW');

                $data['sys_file_reference'][$newId] = $sysFileReference;

                unset($data['sys_file_reference'][$newId]['uid']);

                $data['sys_file_reference'][$newId]['uid_local'] = $replacedByFile->getProperty('uid');

                $pids[] = $sysFileReference['pid'];

                $cmd['sys_file_reference'][$sysFileReference['uid']]['delete'] = 1;
```